lib/elixir/lib/module.ex

```diff
@@ -1015,7 +1040,8 @@ defmodule Module do
   @doc false
   # Used internally to compile types.
   # This function is private and must be used only internally.
-  def store_typespec(module, key, value) when is_atom(key) do
+  @spec store_typespec(module, key :: atom, value :: term) :: Keyword.t
```

Comment:
This function returns `true`, not `Keyword.t`.

Personally I think adding type checking guards to internal functions like this one is too defensive. Normally type check on boundary and use guards for flow control internally.

Is there reason spec'ed this undocumented function and not the other two (get/put attribute)?